Feedback Survey via new Notification Banners

bromiesTM opened the pull request.

bromiesTM committed:
c/fix(MarkdownInlineToken): large spacing between headings and content

Signed-off-by: Kai Henseler <kai.henseler@strato.de>

bromiesTM committed:
c/chore(icons): add new icons

Signed-off-by: Kai Henseler <kai.henseler@strato.de>

bromiesTM committed:
c/style(icons): add icons

Signed-off-by: Kai Henseler <kai.henseler@strato.de>

bromiesTM committed:
c/feat: add notification banner

Refs: PRODAI-255

Co-Authored-By: Thomas Lehmann <t.lehmann@strato.de>

Signed-off-by: Kai Henseler <kai.henseler@strato.de>

bromiesTM committed:
c/feat: add notification banner to app layout

Refs: PRODAI-255

Signed-off-by: Kai Henseler <kai.henseler@strato.de>

bromiesTM committed:
c/docs(.env.sample): add default user role

Spotted while testing the OIDC integration. This required to provision
new users with a user role during first login.

Signed-off-by: Kai Henseler <kai.henseler@strato.de>

thlehmann-ionos commented on backend/open_webui/routers/users.py:
```diff
@@ -347,3 +347,20 @@ async def delete_user_by_id(user_id: str, user=Depends(get_admin_user)):
+############################
+
+
+@router.get("/user/creation", response_model=Optional[int])
```
Can we just return it with the other user data from the `auths` endpoint in `get_session_user()`?

thlehmann-ionos commented on src/lib/IONOS/apis/users.ts:
```diff
@@ -18,3 +18,29 @@ export const deleteAccount = async (token: string): Promise<void> => {
+		})
+		.catch((err) => {
+			console.log(err);
+			error = err.detail;
```
Please don't copy this weird async/Promise cargo cult. This is written in an unnecessarily complicated way.  🫣

Examples for normal implementations `userSignOut()` (core API module) or `deleteAccount()` (custom API module).

thlehmann-ionos commented on src/lib/IONOS/components/icons/EmojiHappy.svelte:
```diff
@@ -0,0 +1,13 @@
+<script lang="ts">
```
Can this commit be squashed with the previous one?

thlehmann-ionos commented on src/lib/IONOS/components/notifications/NotificationManager.svelte:
```diff
@@ -0,0 +1,59 @@
+	import { getUserSettings } from '$lib/apis/users';
+	import { getUserCreation } from '$lib/IONOS/apis/users'
+
+	const i18n = getContext('i18n');
```
Please use a bit stricter typing in "our" components. See other components as examples and hint on linting and typechecking in https://github.com/IONOS-Productivity/gpt-open-webui/blob/live-7/docs/IONOS/customizations.md#linting-and-typechecking

thlehmann-ionos commented on src/lib/IONOS/components/notifications/NotificationManager.svelte:
```diff
@@ -0,0 +1,59 @@
+				label: $i18n.t('Take Our Quick Survey', { ns: 'ionos' }),
+				href: `https://feedback.ionos.com/kujalvoply?l=de&urlVar01=${countryCode}&urlVar02=${userId}&urlVar03=${channels}`,
+				handler: () => {
+					updateSettings({
```
Careful, `updateSettings()` is not doing a deep-merge of the config object, with this line you'd unset any setting in other places that set `{ ionos: { foo: 47 }}`. Please change this in a way that this does not fail now or in the future.

thlehmann-ionos replied:
Fixed by changing `{ ionos: { providedFeedback: true }}` to `{ ionosProvidedFeedback: true }`

thlehmann-ionos commented on docs/IONOS/.env.ionos.sample:
```diff
@@ -3,6 +3,9 @@
 WEBUI_AUTH=True
 WEBUI_SECRET_KEY=generatesomerandomstring
 
+# https://docs.openwebui.com/getting-started/env-configuration/#default_user_role
```
Please add to the commit message this was spotted missing while testing OIDC logins.

thlehmann-ionos reviewed (changes requested):
Please address the review feedback.

* consider changing the account creation date as proposed
* please fix the settings update issue

Apart from that add the issue ID to the commit message as 

```
Refs: <Issue ID>
```

where applicable

thlehmann-ionos commented:
Mind the gap: please adapt the shadow on the header on the Explore page (and maybe on the chat page too as it's not that obvious, but to me there are artifacts visible)

![image](https://github.com/user-attachments/assets/b7345939-c32d-4901-adab-49bb5f23d0ec)

thlehmann-ionos committed:
c/feat(auths): provide created_at of user

We'll use this to present message depending on how long the user account
exists.

Refs: PRODAI-255

thlehmann-ionos committed:
c/feat(Button): add "interactive" prop

Allo buttons to be non-interactive (non-button elements) to use them
within button elements. Apparent buttons within button elements occure
in the agent selector cards.

thlehmann-ionos committed:
c/chore(AgentSelector): fix a11y issue with card not having keypress handler

Fix error:

   A11y: visible, non-interactive elements with an on:click event must
   be accompanied by a keyboard event handler.

This uses the previously added interactive=false prop of Button to make
it a div, to avoid non-compliant button within button.

thlehmann-ionos committed:
c/chore(MessageInput): remove unused function

thlehmann-ionos committed:
c/chore: remove unused imports

thlehmann-ionos committed:
c/chore(InputMenu): fix function type errors

thlehmann-ionos committed:
c/chore(SmallAgentSelector): declare type

To fix type check error

thlehmann-ionos committed:
c/chore(InputMenu): fix type issues of i18n context

thlehmann-ionos committed:
c/chore(InputMenu): removed unused tools rendering

We currently do not configure tools nor does the user have this
functionality. Instead of fixing lots of type issues this code is
removed.

thlehmann-ionos committed:
c/chore(FileItem): fix type issues of i18n context

thlehmann-ionos committed:
c/chore(LoadingCover): fix type for size prop

thlehmann-ionos committed:
c/chore(Scroller): fix type import of ScrollerItem

thlehmann-ionos committed:
c/chore(Spinner): change vendor prefixed to standard mask property

thlehmann-ionos committed:
c/chore(KnowledgeManager): removed undeclared size prop

thlehmann-ionos committed:
c/chore(Account): fix type issue

User can be null here.

thlehmann-ionos committed:
c/chore(account): fix type issue

Declare async function return value as typed Promise

thlehmann-ionos committed:
c/feat: provide config ionos_survey_new_users_url

Refs: PRODAI-255

thlehmann-ionos committed:
c/chore(locales/ionos): download translations

Refs: PRODAI-255
